src/screens/barcodeScan/BarcodeScan.js

```diff
@@ -1,205 +1,426 @@
-            default:
-              break;
-          }
-        }
-        if ("DCA" in elements) { // add driver's license class
-          const licenseClass = elements["DCA"];
-          var classChoice = "";
-          switch (licenseClass) {
-            case "B":
-              classChoice = "2";
-              this.props.updateDriver({id, question:Constants.DLICENSE_CLASS_ID, selection: classChoice});
-              break;
-            case "C":
-              classChoice = "Class C";
-              this.props.updateDriver({id, question:Constants.DLICENSE_CLASS_ID, selection: classChoice});
-              break;
-          }
+import React, { PureComponent } from "react";
+import { StyleSheet, View, PermissionsAndroid } from "react-native";
+import { Button, Spinner, Center } from "native-base";
+import { RNCamera } from "react-native-camera";
+import { updateDriver } from "../../actions/DriverAction";
+import { updateVehicle } from "../../actions/VehicleAction";
+import { connect } from "react-redux";
+import * as Constants from "../../constants";
+import Camera, {
+	Aspect,
+	CaptureQuality,
+	TorchMode,
+} from "react-native-openalpr";
 
-        }
-        this.setState({
-          licenseData: barcode.rawData,
-        });
-        this.props.navigation.goBack();
-      }
-    }
-    return;
-  }
+// Custom styles for views
+const styles = StyleSheet.create({
+	container: {
+		flex: 1,
+		flexDirection: "column",
+		backgroundColor: "black",
+	},
+	preview: {
+		flex: 1,
+		justifyContent: "flex-end",
+		alignItems: "center",
+	},
+	capture: {
+		flex: 0,
+		backgroundColor: "#fff",
+		borderRadius: 5,
+		padding: 15,
+		paddingHorizontal: 20,
+		alignSelf: "center",
+		margin: 20,
+	},
+});
 
-  saveData() {
-    const id = this.props.navigation.state.params.objectID;
-  }
+class BarcodeScan extends PureComponent {
+	constructor(props) {
+		super(props);
+		this.scanDetails = props.navigation.state.params;
+		this.type = this.scanDetails.type;
+		this.questionId = this.scanDetails.questionId;
+		this.questionReducer = this.scanDetails.questionReducer;
+		this.state = {
+			licenseData: "",
+			plate: "",
+			cameraPermissionGranted: false,
+		};
+		this.requireCameraPermission();
+	}
 
-  render() {
+	/*
+	This function checks the device's camera permissions.
+	*/
+	requireCameraPermission = async () => {
+		try {
+			const grantedCamera = await PermissionsAndroid.request(
+				PermissionsAndroid.PERMISSIONS.CAMERA,
+				{
+					title: "Android Camera Permission",
+					message: "Ruina needs access to your camera to scan.",
+					buttonNeutral: "Ask Me Later",
+					buttonNegative: "Cancel",
+					buttonPositive: "Okay",
+				}
+			);
+			if (grantedCamera === PermissionsAndroid.RESULTS.GRANTED) {
+				this.setState({ cameraPermissionGranted: true });
+			} else {
+				// Denied permissions will redirect to form
+				this.setState({ filePermissionsGranted: false });
+			}
+		} catch (err) {
+			console.warn(err);
+		}
+	};
```

Comment:
could this function be moved outside of the render? I found it confusing that it was in the render